File: IT/Input-Output/Esercizi/C++/esercizio25.cpp
/*
file esercizio25.cpp
compilare con: g++ -std=c++11 esercizio25.cpp
*/

#include <iostream>
#include <fstream>
#include <algorithm>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::ifstream;
using std::transform;

int main()
{
	cout << "Apri il file: ";

	string filename;
	getline(cin, filename);

	ifstream f(filename);

	if (f) {
		cout << "Parola da cercare: ";

		string s;
		getline(cin, s);

		string copy = s;
		transform(copy.begin(), copy.end(), copy.begin(), ::tolower);

		bool found = false;
		int i = 1;

		string line;
		while (getline(f, line)) {
			transform(line.begin(), line.end(), line.begin(), ::tolower);
			if (line.find(copy) != -1) {
				if (!found) found = true;
				cout << "La parola \"" << s << "\" compare alla riga " << i << "." << endl;
			}
			i++;
		}

		if (!found) {
			cout << "Parola \"" << s << "\" non trovata!" << endl;
		}
	}
	else {
		cout << "Impossibile aprire il file!" << endl;
	}
}

File: IT/2/Input-Output/Esercizi/Python/esercizio25.py
# file esercizio25.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input

filename = input("Apri il file: ")

try:
	doc = open(filename, "r")
	s = input("Parola da cercare: ")
	found = False
	i = 1

	for line in doc:
		if line.lower().find(s.lower()) != -1:
			if not found: found = True
			print("La parola \"{}\" compare alla riga {}.".format(s, i))
		i = i + 1

	if not found:
		print("Parola \"{}\" non trovata!".format(s))

	doc.close()

except IOError as e:
	print(e)

